integration/tests/cook/util.py

```diff
@@ -1773,6 +1773,8 @@ def has_one_agent():
 def supports_exit_code():
     return using_kubernetes() or is_cook_executor_in_use()
 
+def pool_quota_test_pool():
+    return os.getenv("COOK_TEST_POOL_QUOTA_TEST_POOL", None)
```

Comment:
Instead of requiring the environment variable, how about if we just test in any active pools where the global quota is set to something that's below a certain threshold and also less than our user quota?

Reply:
I can do that. Tradeoff is that it'll be much easier to accidentally disable this test. Tweak a setting and the test silently starts skipping itself.

Reply:
Maybe fail if the number of candidate pools is 0?

Reply:
This whole idea of getting the pool isn't robust. I think we want to exclude the quota test pool from the all_pools so that other dozen tests that use all_pools won't run on it (and potentially break). This is a special pool that we should keep other tests away from, or do we want to have a 'all_pools_and_i_really_mean_all_pools()' function and use that? (IMO: No).

Reply:
Ah, valid point, sounds good